src/super_gradients/training/sg_model/sg_model.py

```diff
@@ -221,11 +221,19 @@ def __init__(self, experiment_name: str, device: str = None, multi_gpu: Union[Mu
     def _set_dataset_properties(self, classes, test_loader, train_loader, valid_loader):
         if any([train_loader, valid_loader, classes]) and not all([train_loader, valid_loader, classes]):
             raise IllegalDataloaderInitialization()
+
         dataset_params = {"batch_size": train_loader.batch_size if train_loader else None,
                           "val_batch_size": valid_loader.batch_size if valid_loader else None,
                           "test_batch_size": test_loader.batch_size if test_loader else None,
                           "dataset_dir": None,
                           "s3_link": None}
+
+        if train_loader and self.multi_gpu == MultiGPUMode.DISTRIBUTED_DATA_PARALLEL:
+            if not all([isinstance(train_loader.sampler, DistributedSampler),
+                        isinstance(valid_loader.sampler, DistributedSampler),
+                        isinstance(test_loader.sampler, DistributedSampler) or test_loader is None]):
```

Comment:
Dod you check this warns..? Im pretty sure isinstance(test_loader.sampler, DistributedSampler) or test_loader is None will through an AttributeError if test sampler is None.